kong/plugins/rate-limiting/policies/init.lua

```diff
@@ -112,6 +135,72 @@ local function get_redis_connection(conf)
 end
 
 
+local function sync_to_redis(premature, conf)
+  if premature then
+    return
+  end
+
+  local red, err = get_redis_connection(conf)
+  if not red then
+    return nil, err
+  end
+
+  red:init_pipeline()
+
+  for cache_key, delta in pairs(cur_delta) do
+    red:eval([[
+      local key, value, expiration = KEYS[1], tonumber(ARGV[1]), ARGV[2]
+      local exists = redis.call("exists", key)
+      redis.call("incrby", key, value)
+      if not exists or exists == 0 then
+        redis.call("expireat", key, expiration)
+      end
+    ]], 1, cache_key, delta, cur_usage[cache_key].expire_at)
+  end
+
+  local _, err = red:commit_pipeline()
+  if err then
+    kong.log.err("failed to commit increment pipeline in Redis: ", err)
+    return nil, err
+  end
+
+  local ok, err = red:set_keepalive(10000, 100)
+  if not ok then
+    kong.log.err("failed to set Redis keepalive: ", err)
+    return nil, err
+  end
+
+  if cur_usage[0] < 500 then
+    --[[
+      if we have less than 500 keys, we can just clear these tables
+      and avoid creating two new tables.
+
+      NOTE: this is a very arbitrary number, so we might choose a better
+      one in the future,
+      now it is like a guard to avoid a large performance downgrade
+      when the number of keys is too large.
+    --]]
+    table_clear(cur_usage)
+    table_clear(cur_delta)
+  else
+    --[[
+      otherwise, we can just create two new tables and let the GC
+      take care of the old tables.
+      because we have a lot of keys, this is more efficient than
+      clearing the tables.
```

Comment:
seems the optimization here is bit hypothetical, i would suggest to just go with a table.clear until we have prove in concrete number